src/polymer-language-server.ts

```diff
@@ -53,14 +58,37 @@ documents.listen(connection);
 // After the server has started the client sends an initilize request. The
 // server receives in the passed params the rootPath of the workspace plus the
 // client capabilites.
-let workspaceRoot: string|null = null;
+let workspaceUri: url.Url|null = null;
+
+function getWorkspaceUri(
+    rootUri: string|null, rootPath: string|null|undefined): url.Url|null {
+  if (rootUri) {
+    return url.parse(rootUri);
+  }
+  if (rootPath) {
+    return url.parse(fileUrl(rootPath));
+  }
+  return null;
+}
+
+function fileUrlToAbsolutePath(fileUrl: url.Url) {
+  const path = decodeURIComponent(fileUrl.pathname!);
+  if (process.platform === 'win32' && path[0] === '/') {
+    return path.slice(1);
+  }
+  return path;
+}
+
 connection.onInitialize((params): InitializeResult => {
-  workspaceRoot = params.rootPath;
-  if (!workspaceRoot) {
+  let maybeWorkspaceUri = getWorkspaceUri(params.rootUri, params.rootPath);
+  // Leave workspaceUri unset if we're initialized in a way we can't handle.
+  if (!maybeWorkspaceUri || maybeWorkspaceUri.protocol !== 'file:') {
     return {capabilities: {}};
   }
+  workspaceUri = maybeWorkspaceUri;

```

Comment:
workspaceUri can be null, I just don't want to set it to a Url with a non-file-based protocol, as the rest of the code assumes that if it's set, it's file:// based.